pkg/git/service.go

```diff
@@ -221,20 +221,65 @@ func setSigningConfig(cfg *ini.File, providerConfig *gitprovider.GitProviderConf
 			return err
 		}
 	case gitprovider.SigningMethodSSH:
+		err := s.configureAllowedSigners(userData.Email, *providerConfig.SigningKey)
+		if err != nil {
+			return err
+		}
+
 		if !cfg.HasSection("gpg") {
 			_, err := cfg.NewSection("gpg")
 			if err != nil {
 				return err
 			}
 		}
-		_, err := cfg.Section("gpg").NewKey("format", "ssh")
+		_, err = cfg.Section("gpg").NewKey("format", "ssh")
+		if err != nil {
+			return err
+		}
+
+		if !cfg.HasSection("gpg \"ssh\"") {
+			_, err := cfg.NewSection("gpg \"ssh\"")
+			if err != nil {
+				return err
+			}
+		}
+
+		allowedSignersFile := filepath.Join(os.Getenv("HOME"), ".config/git/allowed_signers")
+		_, err = cfg.Section("gpg \"ssh\"").NewKey("allowedSignersFile", allowedSignersFile)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func (s *Service) configureAllowedSigners(email, sshKey string) error {
+	homeDir := os.Getenv("HOME")
+	sshDir := filepath.Join(homeDir, ".ssh")
+	allowedSignersFile := filepath.Join(sshDir, "allowed_signers")
+
+	cmd := exec.Command("mkdir", "-p", sshDir)
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	cmd = exec.Command("touch", allowedSignersFile)
```

Comment:
Instead of touch and echo, you can use `os.WriteFile`

Reply:
done